### ISSUE TITLE
Improve Error handling when resolving, unregistering a dependency

### ISSUE THREAD

**Issue by vanlooverenkoen** [vanlooverenkoen/kiwi#32]
```
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown building MyHomePage(dirty, dependencies:
flutter: [_LocalizationsScope-[GlobalKey#6687f], _InheritedTheme], state: _MyHomePageState#b91a6):
flutter: Failed to resolve `String`:
flutter: The type `String` was not registered
flutter: Make sure `String` is added to your KiwiContainer and rerun flutter build
flutter: 'package:kiwi/src/kiwi_container.dart':
flutter: Failed assertion: line 95 pos 7: 'silent || (providers?.containsKey(T) ?? false)'
flutter:
flutter: The relevant error-causing widget was:
flutter:   MyHomePage file:///Users/vanlooverenkoen/Documents/projects/kiwi/flutter_example/lib/main.dart:18:13
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      KiwiContainer.resolve (package:kiwi/src/kiwi_container.dart:95:7)
flutter: #3      _MyHomePageState.build (package:flutter_example/main.dart:59:33)
flutter: #4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4619:28)
flutter: #5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4502:15)
flutter: #6      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
flutter: #7      Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #8      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4481:5)
flutter: #9      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4666:11)
flutter: #10     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4476:5)
flutter: ...     Normal element mounting (24 frames)
flutter: #34     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
flutter: #35     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5947:32)
flutter: ...     Normal element mounting (119 frames)
flutter: #154    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
flutter: #155    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5947:32)
flutter: ...     Normal element mounting (253 frames)
flutter: #408    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
flutter: #409    Element.updateChild (package:flutter/src/widgets/framework.dart:3214:18)
flutter: #410    RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:1148:16)
flutter: #411    RenderObjectToWidgetElement.mount (package:flutter/src/widgets/binding.dart:1119:5)
flutter: #412    RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure> (package:flutter/src/widgets/binding.dart:1061:17)
flutter: #413    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2607:19)
flutter: #414    RenderObjectToWidgetAdapter.attachToRenderTree (package:flutter/src/widgets/binding.dart:1060:13)
flutter: #415    WidgetsBinding.attachRootWidget (package:flutter/src/widgets/binding.dart:941:7)
flutter: #416    WidgetsBinding.scheduleAttachRootWidget.<anonymous closure> (package:flutter/src/widgets/binding.dart:922:7)
flutter: (elided 13 frames from class _AssertionError, class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
flutter:
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════

════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following assertion was thrown building MyHomePage(dirty, dependencies: [_LocalizationsScope-[GlobalKey#6687f], _InheritedTheme], state: _MyHomePageState#b91a6):
Failed to resolve `String`:
The type `String` was not registered
Make sure `String` is added to your KiwiContainer and rerun flutter build
'package:kiwi/src/kiwi_container.dart':
Failed assertion: line 95 pos 7: 'silent || (providers?.containsKey(T) ?? false)'

When the exception was thrown, this was the stack: 
#2      KiwiContainer.resolve (package:kiwi/src/kiwi_container.dart:95:7)
#3      _MyHomePageState.build (package:flutter_example/main.dart:59:33)
#4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4619:28)
#5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4502:15)
#6      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by vanlooverenkoen** [comment]
Adding a KiwiError instead of throwing assert errors. This way it is clear for the developer what is causing the issue. In this case a KiwiError indicates there is something with the kiwi package. By adding more information about the crash and some common solutions it should be clear for the developer what the problem is.
```
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following KiwiError was thrown building ResolveError(dirty):
KiwiError:


Failed to resolve `String`:

The type `String` was not registered

Make sure `String` is added to your KiwiContainer and rerun build_runner build
(If you are using the kiwi_generator)

When using Flutter, most of the time a hot restart is required to setup the KiwiContainer again.

When the exception was thrown, this was the stack: 
#0      KiwiContainer.resolve (package:kiwi/src/kiwi_container.dart:95:7)
#1      ResolveError.build (package:flutter_example/resolve_error_widget.dart:7:21)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:4576:28)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4502:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
Reloaded 1 of 501 libraries in 224ms.
```

**Comment by vanlooverenkoen** [comment]
Fixed in kiwi 1.1.0
